[docs] Add a note about componentWillReceiveProps

dashtinejad opened the pull request:
I've added a complete guide to React Component Lifecycle. I want to improve React documentation much more and this is just the beginning.

Please let me know if I'm doing my job right or not.

lacker commented:
This seems very similar but more-detailed to the content that is in https://facebook.github.io/react/docs/react-component.html right now. Where for each method in the component lifecycle, we have a paragraph describing it. What do you think about just adding extra information there? - it is already a reference page so it would be ok to put more info into it I think.

dashtinejad commented:
@lacker It's a good idea, I'll work on it.

brigand commented:
There are a lot of small issues in this pull request, I commented on a few of them.

dashtinejad commented:
@brigand Thanks for reviewing my documentation. Could you please tell me how can I see your comments? There isn't any comment in the [files](https://github.com/facebook/react/pull/8234/files) for me.

brigand commented:
@dashtinejad  Oops, forgot to press the button.

dashtinejad commented on docs/docs/reference-react-component.md:
```diff
@@ -151,6 +207,28 @@ componentDidMount()
 
 `componentDidMount()` is invoked immediately after a component is mounted. Initialization that requires DOM nodes should go here. If you need to load data from a remote endpoint, this is a good place to instantiate the network request. Setting state in this method will trigger a re-rendering.
 
+If you need to access your element, you can use `ReactDOM.findDOMNode(this)`:
```
I think this note is suitable to be in the reference (we can change it to a Note block).

dashtinejad commented:
@lacker I've combined the 2 documentations, but I'm not sure if it's good or not. As the reference already has the information, I've just moved the examples, screenshots and codepen links there, but I'm not sure if they are **suitable** for reference page.

I've written this documentation, because of when I'm teaching React to my students, they are always confused about Lifecycle, and when I refer them to React Component reference page, they get more confused. So I'd decided to write a more tutorial/visual based documentation than reference one.

Except the 2 notes I've added (I've commented them, so you can find them), personally, I don't think the other parts are suitable for reference.

So, please let me know your decision.

joh1234 commented:
Update README
https://goo.gl/JBljdR

or 
http://bit.ly/2dK6Ojk

lacker commented:
Hmm, I am a bit unclear on what to do from here. It sounds like you don't think the current state of the PR is a good one, but it's all in one doc at least. It doesn't feel right to keep going when you don't actually think it seems good yourself, but it also doesn't feel right to have two different docs aiming at the same thing with a different "style" of communicating it. Maybe this would be better off not as part of the official React docs per se but as a separate page just on Medium or something which is "explaining the lifecycle in a visual way"?

dashtinejad commented:
@lacker Ok, let me commit another revision and just put some notes there. Also good idea about publishing the article on Medium.

gaearon commented:
Thanks! I slightly reworded and merged this in.
